src/Store.cs

```diff
@@ -1,13 +1,165 @@
 using System;
-using System.IO;
-using System.Text;
-using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
 
 namespace Wasmtime
 {
+    /// Represents a Wasmtime store context.
+    public readonly ref struct StoreContext
```

Comment:
FWIW I've avoided exposing this as a first-class type in other bindings because it's in general not safe to use (since it can be used outside of the lifetime of the `Store` itself). I've been adding these methods to `Store` and then having any method that takes a `Store` take anything that you can get a context from, but acquiring the context is just an internal operation.

Reply:
`ref struct` forces the compiler to ensure that this type can only be allocated on a stack frame, so it will make it much harder for a store context to outlive the store (obviously not impossible as you can explicitly dispose a `Store` with the context still stored in the same frame).

I'll see about making `StoreContext` an internal type.